### DISCUSSION TITLE
License Question - Dual License

### DISCUSSION THREAD

**Discussion by Tron731** [rabbitmq/rabbitmq-server#11452] [Questions] [answered]
When I review your marketing materials, your website states that the tool is a ‘dual-license’ Apache 2.0 and MPL 2.0, suggesting to me that I can select either Apache or MPL well using the tool within my product. However,  the attached license file suggests that the license is a mix of MPL, BSD and MIT. I’m not sure if we’re looking in the right spots so if you could please provide us with clarity, it would be greatly appreciated.
[LICENSE.txt](https://github.com/user-attachments/files/15823449/LICENSE.txt)

**Comment by lukebakken** [comment, marked as answer]
Hello -

If you search the RabbitMQ source code, you can see where that license file comes from:

```
git clone https://github.com/rabbitmq/rabbitmq-server.git
cd rabbitmq-server
$ git grep -i 'bsd 2-clause'
packaging/common/LICENSE.tail:25:The BSD 2-Clause license is as follows:
```

OK, so the file is called `LICENSE.tail`. Here is how it is used:

```
$ git grep 'LICENSE\.tail'
Makefile:260:   $(verbose) cat packaging/common/LICENSE.tail >> $@/LICENSE
```

That's interesting, what is in the `Makefile` on line 260:

https://github.com/rabbitmq/rabbitmq-server/blob/main/Makefile#L260-L263

It looks like `LICENSE.tail` is appended to the overall LICENSE file. Most likely, this is due to a **dependency** of RabbitMQ that uses the "BSD 2-clause" license. It does not change RabbitMQ's license at all, which is clearly stated here:

https://github.com/rabbitmq/rabbitmq-server/blob/main/LICENSE

**Reply by lukebakken** [reply]
@Tron731 oh, and I didn't realize that we explicitly suggest using the `rabbitmq-core@` email address here:

https://github.com/rabbitmq/rabbitmq-server/blob/main/LICENSE#L7-L8

Thank you for opening this discussion, because hopefully it will help someone else if they have a question.

**Reply by Tron731** [reply]
Thank you @lukebakken
